src/Readers/ODataReader.v4/ODataCapabilitiesReader.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Annotations;
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Expressions;
+using Microsoft.OData.Edm.Validation;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using Vipr.Core;
+using Vipr.Core.CodeModel;
+
+namespace ODataReader.v4
+{
+    internal static class ODataCapabilitiesReader
+    {
+        private static IEdmModel s_capabilitiesModel;
+        private static readonly IEdmValueTerm s_insertRestrictionsTerm;
+        private static readonly IEdmValueTerm s_updateRestrictionsTerm;
```

Comment:
This is going to turn into a rehash of CapabilitiesHelper from the OData.net folks. We already had some code that parses annotations, why do we need more?